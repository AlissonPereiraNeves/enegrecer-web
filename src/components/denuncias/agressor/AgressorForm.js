import React from 'react';
import Nome from '../../comum/nome';
import CampoTexto from '../../comum/campoTexto';
import Genero from '../../comum/genero';
import Data from '../../comum/data';
import ComboboxEstado from '../../comum/comboboxEstado';
import * as ConstantesCSS from '../ConstantesCss';
import Raca from '../../comum/comboboxRaca';

const AgressorForm = () => (
  <div>
    <h3>Você se sente confortável em dividir conosco informações sobre o agressor?</h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Nunc nisl mi, egestas ac ipsum sit amet,
      blandit posuere metus. Donec imperdiet ut mi sed posuere.
      Pellentesque et varius sapien.
    </p>

    <div className="row">
      <Nome state={'nomeAgressor'} id={'nome-agressor'} />
      <CampoTexto
        state={'cidadeAgressor'}
        id={'cidade-agressor'}
        label={'Cidade'}
        maxLen={40}
        placeholder={''}
        type={'text'}
        divClasse={'input-field col s6'}
      />
    </div>

    <div className="row">
      <Genero
        state={'generoAgressor'}
        id={'genero-agressor'}
        divClasse={'col s6'}
      />

      <Data
        state={'dataNascimentoAgressor'}
        id={'data-nascimento-agressor'}
        label={'Data de Nascimento'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s6`}
      />
    </div>

    <div className="row">
      <Raca id={'raca-agressor'} state={'racaAgressor'} divClasse={'col s6'} somenteRacasVitima={false} />
      <CampoTexto
        state={'bairroAgressor'}
        id={'bairro-agressor'}
        label={'Bairro'}
        maxLen={100}
        placeholder={''}
        type={'text'}
        divClasse={'input-field col s6'}
      />
    </div>

    <div className="row">
      <CampoTexto
        state={'descricaoAgressor'}
        id={'descricao-agressor'}
        label={'Descrição do agressor'}
        maxLen={100}
        placeholder={''}
        type={'text'}
        divClasse={'input-field col s12'}
      />
      <ComboboxEstado
        state={'estadoAgressor'}
        id={'estado-agressor'}
        classes={'col s6'}
      />
    </div>
  </div>
);

export default AgressorForm;
