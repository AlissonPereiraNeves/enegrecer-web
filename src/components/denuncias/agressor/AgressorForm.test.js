import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import AgressorForm from './AgressorForm';

describe('AgressorForm', () => {
  it('deve seguir a estrutura definida ', () => {
    const agressorForm = shallow(<AgressorForm />);
    const agressorFormJson = toJson(agressorForm);
    expect(agressorFormJson).toMatchSnapshot();
  });
});

