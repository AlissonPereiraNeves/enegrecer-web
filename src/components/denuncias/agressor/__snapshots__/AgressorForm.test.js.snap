// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgressorForm deve seguir a estrutura definida  1`] = `
<div>
  <div
    className="row"
  >
    <nome
      id="nome-agressor"
      onChange={[Function]}
      state="nomeAgressor"
      validacoes={Array []}
    />
    <genero
      divClasse="col s6"
      id="genero-agressor"
      state="generoAgressor"
    />
  </div>
  <div
    className="row"
  >
    <comboEstado
      divClasse="col s6"
      id="estado-agressor"
      state="estadoAgressor"
    />
    <combobox
      divClasse="col s6"
      id="cidade-agressor"
      itens={
        Array [
          "Belo Horizonte",
        ]
      }
      label="Cidade *"
      state="cidadeAgressor"
      valorPadrao=""
    />
  </div>
  <div
    className="row"
  >
    <Data
      divClasse="input-field col s6"
      id="data-nascimento-agressor"
      label="Data de Nascimento"
      state="dataNascimentoAgressor"
    />
    <comboRaca
      divClasse="col s6"
      id="raca-agressor"
      somenteRacasVitima={false}
      state="racaAgressor"
    />
  </div>
  <div
    className="row"
  >
    <campoTexto
      divClasse="input-field col s6"
      id="bairro-agressor"
      label="Bairro"
      maxLen={100}
      placeholder=""
      state="bairroAgressor"
      type="text"
    />
  </div>
  <div
    className="row"
  >
    <campoTexto
      divClasse="input-field col s12"
      id="descricao-agressor"
      label="Descrição do agressor"
      maxLen={100}
      placeholder=""
      state="descricaoAgressor"
      type="text"
      validacoes={
        Array [
          [Function],
        ]
      }
    />
  </div>
</div>
`;
